### PR TITLE
fix manufacturer bug #365

### LINKED ISSUE #365: fix bugs in manufacturers
Fix issue @VKTB found with manufacturers where you create, delete . then go to create a new manufacturer

### PR DESCRIPTION
**Pull request by MatteoGuarnaccia5**
## Description

Added prop in delete dialog for manufacturers to set the selected manufacturer back to undefined. So that it does not try to fetch a manufacturer to populate the add/edit dialog

## Testing instructions

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking

closes #365

### CODE REVIEW

**Review by joelvdavies — COMMENTED** [review]
Would removing
```
  const { data: selectedManufacturerData } = useManufacturer(
    selectedManufacturer?.id
  );
```
In the manufacturer dialog and using the `selectedManufacturer` in the manufacturerDialog like we have done elsewhere resolve the issue? It would mean it no longer tries to look them up again and the delete will clear the cache anyway so it will simply be reset to null without having to introduce a new prop.

File: public/inventory-management-system-settings.example.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     },
     {
       "section": "Inventory Management System",
-      "link": "/manufacturer",
+      "link": "/manufacturers",
       "displayName": "Manufacturer",
```
Comment (joshuadkitenge) [inline public/inventory-management-system-settings.example.json:25]:
```suggestion
      "displayName": "Manufacturers",
```

File: craco.config.js [hunk 2]
```diff
@@ -2,20 +2,16 @@ module.exports = {
   webpack: {
```
Comment (joshuadkitenge) [inline craco.config.js:2]: un commit this file please

File: src/manufacturer/manufacturerDialog.component.tsx [hunk 3]
```diff
@@ -207,45 +200,45 @@ function ManufacturerDialog(props: ManufacturerDialogProps) {
...
+    if (manufacturerDetails && selectedManufacturer) {
       const hasErrors = handleErrors();
 
       if (hasErrors) {
         return;
       }
 
       const isNameUpdated =
-        manufacturerDetails.name !== selectedManufacturerData.name;
+        manufacturerDetails.name !== selectedManufacturer?.name;
```
Comment (joelvdavies) [inline src/manufacturer/manufacturerDialog.component.tsx:211]:
```suggestion
        manufacturerDetails.name !== selectedManufacturer.name;
```
Same for the others below, line 203 ensures it isn't undefined.

**Review by joelvdavies — COMMENTED** [review]
One minor thing

**Review by joelvdavies — APPROVED** [review]

**Review by joshuadkitenge — APPROVED** [review]

**Pull request merged**